src/utils/sequelize-method.ts

```diff
@@ -0,0 +1,135 @@
+import isObject from 'lodash/isObject';
+import type { Op, WhereOperators, WhereLeftOperand } from '..';
+
+/**
+ * Utility functions for representing SQL functions, and columns that should be escaped.
+ * Please do not use these functions directly, use Sequelize.fn and Sequelize.col instead.
+ *
+ * @private
+ */
+export class SequelizeMethod {}
+
+/**
+ * Do not use me directly. Use {@link Sequelize.fn}
+ */
+export class Fn extends SequelizeMethod {
+  private readonly fn: string;
+  private readonly args: unknown[];
+
+  constructor(fn: string, args: unknown[]) {
+    super();
+    this.fn = fn;
+    this.args = args;
+  }
+
+  clone(): Fn {
+    return new Fn(this.fn, this.args);
+  }
+}
+
+/**
+ * Do not use me directly. Use {@link Sequelize.col}
+ */
+export class Col extends SequelizeMethod {
+  private readonly col: string[] | string;
+
+  constructor(col: string[] | string, ...args: string[]) {
+    super();
+    // TODO(ephys): this does not look right. First parameter is ignored if a second parameter is provided.
```

Comment:
This is for a future PR, right?

Reply:
Yep! It's in my backlog but something I want to take a look at in a different PR